File: Readio/src/main/java/ninegle/Readio/user/domain/RefreshToken.java
package ninegle.Readio.user.domain;

import java.time.LocalDateTime;

import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class RefreshToken {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String refreshToken;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id")
	//단방향으로 유저와 연결
	private User user;

	@UpdateTimestamp
	private LocalDateTime UpdateTimeAt;

	@Builder
	public RefreshToken(String refreshToken, User user) {
		this.refreshToken = refreshToken;
		this.user = user;
	}

	public void newSetRefreshToken(String refreshToken) {
		this.refreshToken = refreshToken;//갱신 시간 업데이트 (옵션)
	}
}

File: Readio/src/main/java/ninegle/Readio/user/domain/RefreshToken.kt
package ninegle.Readio.user.domain

import jakarta.persistence.*
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime


@Entity
@Table(name = "refresh_token")
class RefreshToken (
    @field:Column(nullable = false)
    var refreshToken: String, //단방향으로 유저와 연결


    @field:JoinColumn(name = "user_id") @field:ManyToOne(fetch = FetchType.LAZY)
    var user: User) {

    constructor() : this("", User()) // 기본 생성자 (JPA용) }


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null

    @UpdateTimestamp
    private var UpdateTimeAt: LocalDateTime? = null

    fun newSetRefreshToken(refreshToken: String) {
        this.refreshToken = refreshToken //갱신 시간 업데이트 (옵션)
    }
}
